integration-tests/src/main/java/org/apache/druid/testing/clients/CompactionResourceTestClient.java

```diff
@@ -0,0 +1,179 @@
+
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      throw new ISE(
+          "Error while deleting compaction config status[%s] content[%s]",
+          response.getStatus(),
+          response.getContent()
+      );
+    }
+  }
+
+  public CoordinatorCompactionConfig getCoordinatorCompactionConfigs() throws Exception
+  {
+    String url = StringUtils.format("%sconfig/compaction", getCoordinatorURL());
+    StatusResponseHolder response = httpClient.go(
+        new Request(HttpMethod.GET, new URL(url)), responseHandler
+    ).get();
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      throw new ISE(
+          "Error while getting compaction config status[%s] content[%s]",
+          response.getStatus(),
+          response.getContent()
+      );
+    }
+    return jsonMapper.readValue(response.getContent(), new TypeReference<CoordinatorCompactionConfig>() {});
+  }
+
+  public DataSourceCompactionConfig getDataSourceCompactionConfig(String dataSource) throws Exception
+  {
+    String url = StringUtils.format("%sconfig/compaction/%s", getCoordinatorURL(), StringUtils.urlEncode(dataSource));
+    StatusResponseHolder response = httpClient.go(
+        new Request(HttpMethod.GET, new URL(url)), responseHandler
+    ).get();
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      throw new ISE(
+          "Error while getting compaction config status[%s] content[%s]",
+          response.getStatus(),
+          response.getContent()
+      );
+    }
+    return jsonMapper.readValue(response.getContent(), new TypeReference<DataSourceCompactionConfig>() {});
+  }
+
+  public void forceTriggerAutoCompaction() throws Exception
+  {
+    String url = StringUtils.format("%scompaction/compact", getCoordinatorURL());
+    StatusResponseHolder response = httpClient.go(new Request(HttpMethod.POST, new URL(url)), responseHandler).get();
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      throw new ISE(
+          "Error while force trigger auto compaction status[%s] content[%s]",
+          response.getStatus(),
+          response.getContent()
+      );
+    }
+  }
+
+  public void updateCompactionTaskSlot(Double compactionTaskSlotRatio, Integer maxCompactionTaskSlots) throws Exception
+  {
+    String url = StringUtils.format("%sconfig/compaction/taskslots?ratio=%s&max=%s",
+                                    getCoordinatorURL(),
+                                    StringUtils.urlEncode(compactionTaskSlotRatio.toString()),
+                                    StringUtils.urlEncode(maxCompactionTaskSlots.toString()));
+    StatusResponseHolder response = httpClient.go(new Request(HttpMethod.POST, new URL(url)), responseHandler).get();
+    if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+      throw new ISE(
+          "Error while force trigger auto compaction status[%s] content[%s]",
```

Comment:
Should the message say something about updating compaction task slots instead?

Reply:
Done